contracts/abstracts/Controlable.sol

```diff
@@ -43,15 +43,17 @@ abstract contract Controlable is AccessControlUpgradeable {
     _;
   }
 
-  function _changeSupportedContract(address contractAddress) internal onlyAdmin returns (bool success) {
-    if (supportedContracts[contractAddress]) {
-      // Le contrat est déjà pris en charge, donc on le supprime
-      supportedContracts[contractAddress] = false;
-      emit SupportedContractRemoved(contractAddress);
+  function _changeSupportedContract(address contractAddress, bool isSupported) internal returns (bool success) {
+    if (isSupported == true) {
+      if (!_isSupportedContract(contractAddress)) {
```

Comment:
There is no need for this line (in both branch) we don't need to care about the previous state, since there is now a bool as argument, just save the bool as the new value of supportedContracts[contractAddress] and return the event base o the value of is supported.